Add driver for the Futaba TOSD-5711BB VFDisplay on Elonex Artisan/Sca…

ajw107 opened the pull request:
…leo Media Centre PCs
futaba.so driver and LCDd.conf change for the Futaba TOSD-5711BB VFD

ajw107 committed:
Add driver for the Futaba TOSD-5711BB VFDisplay on Elonex Artisan/Scaleo Media Centre PCs

haraldg commented on server/drivers/futaba.c:
```diff
@@ -0,0 +1,619 @@
+	futaba_start_driver(drvthis);
+	memset(p->framebuf, ' ', p->width * p->height);
+	memcpy(p->old_framebuf, p->framebuf,p->width * p->height);
+	report(RPT_DEBUG, "%s: init() done", drvthis->name);
```
Use debug() here, so this gets optimized away on non-debug builds ...
The same is true at more places.

haraldg commented on server/drivers/futaba.c:
```diff
@@ -0,0 +1,619 @@
+	//PrivateData *p = drvthis->private_data;
+	debug(RPT_DEBUG, "Backlight %s command ignored as not supported", (on) ? "ON" : "OFF");
+}
+
```
The last two functions are not mandatory. Maybe just leave them out completely.

haraldg commented on server/drivers/futaba.c:
```diff
@@ -0,0 +1,619 @@
+#include <stdint.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <libusb-1.0/libusb.h>
```
This line fails on travis build. Drop the libusb-1.0 part.

haraldg commented on acinclude.m4:
```diff
@@ -136,6 +136,10 @@ dnl				else
+                futaba)
+                        DRIVERS="$DRIVERS futaba${SO}"
+                        actdrivers=["$actdrivers futaba"]
+                        ;;
```
You should check if libusb is actually available. See how it is done for the picolcd driver for example.

haraldg reviewed (changes requested):
This looks quite good. The only thing that is missing is the section describing this driver in the user guide (even if it will be quite trivial) and the LCDd man page. See http://lcdproc.sourceforge.net/docs/current-dev.html#documentation also about coding style. How do you want to be listed in the list of contributors?

ajw107 commented:
Thank you so much for going through everything.  I’ll definitely make the changes you suggest and see how to update the documentation as well.  I’ll issue another pull request when I’m done.
Alex

From: haraldg [mailto:notifications@github.com]
Sent: 19 December 2016 21:32
To: lcdproc/lcdproc <lcdproc@noreply.github.com>
Cc: Alexander Wood <alex@alex-wood.org.uk>; Author <author@noreply.github.com>
Subject: Re: [lcdproc/lcdproc] Add driver for the Futaba TOSD-5711BB VFDisplay on Elonex Artisan/Sca… (#33)


@haraldg requested changes on this pull request.

This looks quite good. The only thing that is missing is the section describing this driver in the user guide (even if it will be quite trivial) and the LCDd man page. See http://lcdproc.sourceforge.net/docs/current-dev.html#documentation also about coding style. How do you want to be listed in the list of contributors?

________________________________

In server/drivers/futaba.c<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-13641034>:

> +     if (p->framebuf == NULL) {

+              report(RPT_ERR, "%s: unable to create framebuffer", drvthis->name);

+              return -1;

+       }

+

+       p->old_framebuf = (char *)malloc(p->width * p->height);

+       if (p->old_framebuf == NULL) {

+              report(RPT_ERR, "%s: unable to create old_framebuffer", drvthis->name);

+              return -1;

+       }

+

+       /* write to the file */

+       futaba_start_driver(drvthis);

+       memset(p->framebuf, ' ', p->width * p->height);

+       memcpy(p->old_framebuf, p->framebuf,p->width * p->height);

+       report(RPT_DEBUG, "%s: init() done", drvthis->name);

Use debug() here, so this gets optimized away on non-debug builds ...
The same is true at more places.

________________________________

In server/drivers/futaba.c<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-13641034>:

> +     debug(RPT_DEBUG, "Contrast: %d command ignored as not supported", promille);

+}

+

+/**

+ * Turn the display backlight on or off.

+ * Display is LED so we ignore it.

+ * \param drvthis  Pointer to driver structure.

+ * \param on       New backlight status.

+ */

+MODULE_EXPORT void

+futaba_backlight (Driver *drvthis, int on)

+{

+       //PrivateData *p = drvthis->private_data;

+       debug(RPT_DEBUG, "Backlight %s command ignored as not supported", (on) ? "ON" : "OFF");

+}

+

The last two functions are not mandatory. Maybe just leave them out completely.

________________________________

In server/drivers/futaba.c<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-13641034>:

> + *

+ * You should have received a copy of the GNU General Public License

+ * along with this program; if not, write to the Free Software

+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301

+ *

+ */

+

+#ifdef HAVE_CONFIG_H

+#include "config.h"

+#endif

+

+#include <stdio.h>

+#include <stdint.h>

+#include <unistd.h>

+#include <stdlib.h>

+#include <libusb-1.0/libusb.h>

This line fails on travis build. Drop the libusb-1.0 part.

________________________________

In acinclude.m4<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-13641034>:

> @@ -136,6 +136,10 @@ dnl                           else

                       DRIVERS="$DRIVERS EyeboxOne${SO}"

                       actdrivers=["$actdrivers EyeboxOne"]

                       ;;

+                futaba)

+                        DRIVERS="$DRIVERS futaba${SO}"

+                        actdrivers=["$actdrivers futaba"]

+                        ;;

You should check if libusb is actually available. See how it is done for the picolcd driver for example.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-13641034>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AB-KPNvdtCDzXyc8bTapzV-3kC5S4U5Oks5rJvfDgaJpZM4LQmqD>.

ajw107 committed:
Multiple corrections to bring driver in line with lcdproc code guidlines:
1) Removed both functions overrides for controlling contrast and back-lighting, these were empty functions in anycase as the display does not support contrast or back-lighting
2) Changed include for libusb-1.0, and moved it into it's proper place in futaba.h
3) Changed global include define in futaba.h to the more correct FUTABA_H
4) acinclude now checks for libusb (0.1 and 1.0) and checks for either of them during configure
5) Massive re-write to init and send functions to choose the right commands during compile depending if libusb 1.0 or 0.1 is installed (preferring 1.0 if both are installed) using picolcd as a template
   where 1.0 and 0.1 functions are transplantable, defines exist in futaba.h to use the correct one
6) Changed a lot of debug messages to use the debug function and not the report function, so that they are only compiled in with the --debug option.  Also added a lot more debug output if required.
7) Icon array size was incorrect, and also using plain int, which caused compile time warnings when bit-shifting as no bounds, so changed to uint64_t (as 40 values min)
8) Added a slowed down retry step when resetting the display during init, as sometyimes can happen too fast

haraldg commented on server/drivers/futaba.c:
```diff
@@ -167,7 +187,7 @@ MODULE_EXPORT char *symbol_prefix = "futaba_";
 				if( futabaSendReport(p->my_handle, &my_report) )
 				{
-					fprintf(stderr,"Failed in futabaSendString()\n");
+					debug(RPT_ERR,"[%s] Failed to send report", drvthis->name);
```
Are you sure you want debug() here and not report()?

haraldg commented on server/drivers/futaba.c:
```diff
@@ -307,17 +460,23 @@ futaba_init (Driver *drvthis)
 	p = (PrivateData *) calloc(1, sizeof(PrivateData));
 	if (p == NULL)
+	{
+		report(RPT_ERR, "%s: Private Data is NULL", drvthis->name);
```
This message is somewhat misleading. We are simply out of memory. (Should never happen on a modern system due to overcommitting by the kernel.)

haraldg reviewed (commented):
This is looking good, found only two minor nits.

Since libusb0.1 is obsolete, I wouldn't have put effort into supporting it for a new driver, but of course it's a nice thing to have. This prompts the question: Do you feel motivated to work on issue #13? - You appear to be quite competent for that kind of work... ;)

Do you need any help with documentation?

ajw107 commented:
Thanks for the pointers, very much appreciated.  I’ve almost finished the documentation (I think, famous last words) so should be able to do another push soon.
Alex

From: haraldg [mailto:notifications@github.com]
Sent: 22 December 2016 21:56
To: lcdproc/lcdproc <lcdproc@noreply.github.com>
Cc: Alexander Wood <alex@alex-wood.org.uk>; Author <author@noreply.github.com>
Subject: Re: [lcdproc/lcdproc] Add driver for the Futaba TOSD-5711BB VFDisplay on Elonex Artisan/Sca… (#33)


@haraldg commented on this pull request.

This is looking good, found only two minor nits.

Since libusb0.1 is obsolete, I wouldn't have put effort into supporting it for a new driver, but of course it's a nice thing to have. This prompts the question: Do you feel motivated to work on issue #13<https://github.com/lcdproc/lcdproc/issues/13>? - You appear to be quite competent for that kind of work... ;)

Do you need any help with documentation?

________________________________

In server/drivers/futaba.c<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-14232924>:

> @@ -167,7 +187,7 @@ MODULE_EXPORT char *symbol_prefix = "futaba_";

                       {

                               if( futabaSendReport(p->my_handle, &my_report) )

                               {

-                                     fprintf(stderr,"Failed in futabaSendString()\n");

+                                     debug(RPT_ERR,"[%s] Failed to send report", drvthis->name);

Are you sure you want debug() here and not report()?

________________________________

In server/drivers/futaba.c<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-14232924>:

> @@ -307,17 +460,23 @@ futaba_init (Driver *drvthis)

        /* Allocate and store private data */

        p = (PrivateData *) calloc(1, sizeof(PrivateData));

        if (p == NULL)

+       {

+              report(RPT_ERR, "%s: Private Data is NULL", drvthis->name);

This message is somewhat misleading. We are simply out of memory. (Should never happen on a modern system due to overcommitting by the kernel.)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-14232924>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AB-KPGrOsqaCjawdnAm8g16r8YJ4eEqjks5rKvHzgaJpZM4LQmqD>.

ajw107 commented:
Hi
Sorry to bother you, but I’m having problems doing the docbook documentation.  I have to admit to never using docbook before, and only limited experience with xml, could you suggest an editor I could use?  Preferably as wysiwyg as possible, he he.
Alex

From: haraldg [mailto:notifications@github.com]
Sent: 22 December 2016 21:56
To: lcdproc/lcdproc <lcdproc@noreply.github.com>
Cc: Alexander Wood <alex@alex-wood.org.uk>; Author <author@noreply.github.com>
Subject: Re: [lcdproc/lcdproc] Add driver for the Futaba TOSD-5711BB VFDisplay on Elonex Artisan/Sca… (#33)


@haraldg commented on this pull request.

This is looking good, found only two minor nits.

Since libusb0.1 is obsolete, I wouldn't have put effort into supporting it for a new driver, but of course it's a nice thing to have. This prompts the question: Do you feel motivated to work on issue #13<https://github.com/lcdproc/lcdproc/issues/13>? - You appear to be quite competent for that kind of work... ;)

Do you need any help with documentation?

________________________________

In server/drivers/futaba.c<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-14232924>:

> @@ -167,7 +187,7 @@ MODULE_EXPORT char *symbol_prefix = "futaba_";

                       {

                               if( futabaSendReport(p->my_handle, &my_report) )

                               {

-                                     fprintf(stderr,"Failed in futabaSendString()\n");

+                                     debug(RPT_ERR,"[%s] Failed to send report", drvthis->name);

Are you sure you want debug() here and not report()?

________________________________

In server/drivers/futaba.c<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-14232924>:

> @@ -307,17 +460,23 @@ futaba_init (Driver *drvthis)

        /* Allocate and store private data */

        p = (PrivateData *) calloc(1, sizeof(PrivateData));

        if (p == NULL)

+       {

+              report(RPT_ERR, "%s: Private Data is NULL", drvthis->name);

This message is somewhat misleading. We are simply out of memory. (Should never happen on a modern system due to overcommitting by the kernel.)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-14232924>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AB-KPGrOsqaCjawdnAm8g16r8YJ4eEqjks5rKvHzgaJpZM4LQmqD>.

haraldg commented:
> Sorry to bother you, but I’m having problems doing the docbook documentation. I have to admit to never using docbook before, and only limited experience with xml, could you suggest an editor I could use? Preferably as wysiwyg as possible, he he.

Writing my driver for lcdproc was my first encounter with docbook too ... I just looked at documentation of other drivers to use as templates and used the dev-guide for guidance as which files need changing. I just used vim.

I suggest you push whatever you come up with and I'll comment on your code. That's probably the easiest way. Just write here in the PR what your are still unhappy with. When we are done we will squash all commits into one to have a clean and complete commit in the repository.

ajw107 committed:
-Improved function descriptions
-Aligned more with lcdproc code style, function name changes
-GNU indent format (first time used, agh)
-Improved comments here and in .h file
-Changed RPT_ERR to RPT_WARNINGerr if an error occurs, but
     code still continesg on
-Added a few todo's for future reference
-Removed goto in init_driver, never comfortable with it
-Removed global futaba driver struct and added to private data
-Added alternate usb config for some reason (picolcd has it, but
     comments it never works)
-Configured LIBUSB 1.0 to give log messages
-Use futaba_shutdown (maybe futaba_close) instead of many
     discrete libusb close statements

haraldg commented on docs/lcdproc-user/drivers/futaba.docbook:
```diff
@@ -0,0 +1,140 @@
+    <para>The driver is based on the <ulink
+    url="http://libusb.info/"><filename>libusb</filename></ulink> USB library
+    (Versions 1.0 and 0.1), which should make it work with Linux, the
+    different BSB variants as well as Darwin/MacOS X.</para>
```
BSB = BSD?

haraldg commented on docs/lcdproc-user/drivers/futaba.docbook:
```diff
@@ -0,0 +1,140 @@
+      <para>When using a <filename>libusb</filename> based driver like
+      <code>futaba</code>, <application>LCDd</application> needs to be started
+      as root.</para>
+    </note>
```
Will dropping priviledges after initialization (the usual LCDd way) work or has LCDd run as root all the time?

haraldg commented on docs/lcdproc-user/drivers/futaba.docbook:
```diff
@@ -0,0 +1,140 @@
+            <para>Should not be changed from the default of 7x1.</para>
+          </listitem>
+        </varlistentry>
+
```
Looks like the size parameters are actually hardcoded in the driver and such an entry in the configuration file won't have any effect anyway. Just drop this from the documentation and LCDd.conf

haraldg commented on docs/lcdproc-user/drivers/futaba.docbook:
```diff
@@ -0,0 +1,140 @@
+    get these to work, with three settings: one to enable/disable flashing;
+    one to set the flash rate; and one to choose between ':' (colon) or '.'
+    (dot).</para>
+  </sect2>
```
I have a smilar situation in the olimex_1x9 driver. Maybe you can take some inspiration from there or maybe you come up with something better, for me to copy ... ;)

haraldg commented on server/drivers/futaba.c:
```diff
@@ -78,21 +93,22 @@
 #define PRODUCT_ID 0x7000
-futabaDriver_t g_futabaDriver;
+//futabaDriver_t g_futabaDriver;
 
```
Is there any point in keeping this line?

haraldg commented on server/drivers/futaba.c:
```diff
@@ -207,292 +220,319 @@ MODULE_EXPORT char *symbol_prefix = "futaba_";
+			break;
+		}
 	}
-done:
```
IMO the goto was fine. Anyway, it's your code, so you decide.

haraldg commented on docs/lcdproc-user/drivers/futaba.docbook:
```diff
@@ -0,0 +1,140 @@
+        <varlistentry>
+          <term><property>TitleSpeed</property> =
+          <parameter><replaceable>TITLESPEED</replaceable></parameter></term>
+
```
This is a parameter of the server, not the driver and need not be documented here. Also drop it from LCDd.conf together with the other server parameters. I missed that in the previous reviews.

haraldg reviewed (commented):
This is going to be the perfect driver, thanks a lot for all the effort you put into this.

You need to add your new docbook file to lcdproc-user.docbook and drivers.docbook to get it actually included. 

After fixing the LCDd.conf issue, that I missed previously, this is ready to merge as far as I'm concerned. You can either squash and force-push into this PR or push the last fixes directly and I'll do the squashing on github.

ajw107 committed:
Updates to the documentation for the futaba driver.

ajw107 commented:
Hi
Just about to submit the final commit (well until I add extra features).  Thank you so much for all of your help, especially over the Christmas, an insane time of year, so mad in fact I can’t remember making the last commit and was shocked when I saw your email, he he…

I’ve made the changes you suggested, I must apologise for the errors in the docbook files, not only did I forgot to add the other auxiliary files, but my copy and pasting from the picolcd driver documentation got me into trouble with a few things.  You are right, the driver does not need to be run as root permanently, and I also took out the stuff about usbfs, as this is obviously old info (do you want me to do the same for the picolcd drivers docs)?  I’ll also supmit updated docs when the linux patch and script.xbmc.lcdproc patches are accepted.

On the LCDd.conf I removed everything from the futaba section, but I’ve added a note to the documentation about the TitleSpeed, mainly because I remember being very confused when I used the driver and everything zoomed past so quick and it taking me ages to find the property (although that may just be me, he he).

Sorry for leaving in one of commented out lines of code, at work we’re not allowed to delete any lines, and I tend to get into the habit of it.  I’ve also left the goto code using a flag instead to control the program flow, blame my programming teach at uni who had a serious thing against them and drummed it into us all, he he.

Finally, I’ve updated the Changelog.md and CREDITS.md files, apologies if this is not my place to do so, just thought it may be easier for you.  Oh, and could you squash my code into one commit, I ask as I’ve not actually done that before, and I have this horrible image of me accidently reverting it all by mistake, he he.
Yours
Alex

From: haraldg [mailto:notifications@github.com]
Sent: 25 December 2016 16:00
To: lcdproc/lcdproc <lcdproc@noreply.github.com>
Cc: Alexander Wood <alex@alex-wood.org.uk>; Author <author@noreply.github.com>
Subject: Re: [lcdproc/lcdproc] Add driver for the Futaba TOSD-5711BB VFDisplay on Elonex Artisan/Sca… (#33)


@haraldg commented on this pull request.

This is going to be the perfect driver, thanks a lot for all the effort you put into this.

You need to add your new docbook file to lcdproc-user.docbook and drivers.docbook to get it actually included.

After fixing the LCDd.conf issue, that I missed previously, this is ready to merge as far as I'm concerned. You can either squash and force-push into this PR or push the last fixes directly and I'll do the squashing on github.

________________________________

In docs/lcdproc-user/drivers/futaba.docbook<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-14365313>:

> +          <para>The patched version of

+          <application>scripts.xbmc.lcproc</application> can be found <ulink

+          url="https://github.com/ajw107/script.xbmc.lcdproc">here</ulink>,

+          this should be added upstream soon too.</para>

+        </listitem>

+      </itemizedlist>

+    </note>

+  </sect2>

+

+  <sect2 id="futaba-requirements">

+    <title>Requirements</title>

+

+    <para>The driver is based on the <ulink

+    url="http://libusb.info/"><filename>libusb</filename></ulink> USB library

+    (Versions 1.0 and 0.1), which should make it work with Linux, the

+    different BSB variants as well as Darwin/MacOS X.</para>

BSB = BSD?

________________________________

In docs/lcdproc-user/drivers/futaba.docbook<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-14365313>:

> +    </note>

+  </sect2>

+

+  <sect2 id="futaba-requirements">

+    <title>Requirements</title>

+

+    <para>The driver is based on the <ulink

+    url="http://libusb.info/"><filename>libusb</filename></ulink> USB library

+    (Versions 1.0 and 0.1), which should make it work with Linux, the

+    different BSB variants as well as Darwin/MacOS X.</para>

+

+    <note>

+      <para>When using a <filename>libusb</filename> based driver like

+      <code>futaba</code>, <application>LCDd</application> needs to be started

+      as root.</para>

+    </note>

Will dropping priviledges after initialization (the usual LCDd way) work or has LCDd run as root all the time?

________________________________

In docs/lcdproc-user/drivers/futaba.docbook<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-14365313>:

> +    <title>Configuration in LCDd.conf</title>

+

+    <sect3 id="futaba-config-section">

+      <title>[futaba]</title>

+

+      <variablelist>

+

+        <varlistentry>

+          <term><property>Size</property> =

+          <parameter>WIDTHxHEIGHT</parameter></term>

+

+          <listitem>

+            <para>Should not be changed from the default of 7x1.</para>

+          </listitem>

+        </varlistentry>

+

Looks like the size parameters are actually hardcoded in the driver and such an entry in the configuration file won't have any effect anyway. Just drop this from the documentation and LCDd.conf

________________________________

In docs/lcdproc-user/drivers/futaba.docbook<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-14365313>:

> +<para>There are no Brightness, Contrast or Back-light controls as

+the display does not support these functions</para>

+</note>

+

+    </sect3>

+  </sect2>

+

+  <sect2 id="futaba-status">

+    <title>futaba driver status</title>

+

+    <para>All icons/indicators function well. Currently the ':' (colon) or '.'

+    (dot) in between characters do NOT operate. In a future update I hope to

+    get these to work, with three settings: one to enable/disable flashing;

+    one to set the flash rate; and one to choose between ':' (colon) or '.'

+    (dot).</para>

+  </sect2>

I have a smilar situation in the olimex_1x9 driver. Maybe you can take some inspiration from there or maybe you come up with something better, for me to copy ... ;)

________________________________

In server/drivers/futaba.c<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-14365313>:

> @@ -78,21 +93,22 @@



 #define VENDOR_ID 0x0547

 #define PRODUCT_ID 0x7000

-futabaDriver_t g_futabaDriver;

+//futabaDriver_t g_futabaDriver;



Is there any point in keeping this line?

________________________________

In server/drivers/futaba.c<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-14365313>:

>       }

-done:

IMO the goto was fine. Anyway, it's your code, so you decide.

________________________________

In docs/lcdproc-user/drivers/futaba.docbook<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-14365313>:

> +

+      <variablelist>

+

+        <varlistentry>

+          <term><property>Size</property> =

+          <parameter>WIDTHxHEIGHT</parameter></term>

+

+          <listitem>

+            <para>Should not be changed from the default of 7x1.</para>

+          </listitem>

+        </varlistentry>

+

+        <varlistentry>

+          <term><property>TitleSpeed</property> =

+          <parameter><replaceable>TITLESPEED</replaceable></parameter></term>

+

This is a parameter of the server, not the driver and need not be documented here. Also drop it from LCDd.conf together with the other server parameters. I missed that in the previous reviews.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/lcdproc/lcdproc/pull/33#pullrequestreview-14365313>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AB-KPG0nx_qVK3S_EwFXL5y39U4Xf1bEks5rLpL3gaJpZM4LQmqD>.

haraldg commented:
No need to be sorry or apologize for anything. This is good stuff and we all learn as we go and reviews almost always find ways to improve the code - which is why it is a good thing to have them. (Maybe I'm a bit biased from working on the linux kernel though.)

You seem to be reasonable and competent. We probably should give you commit access if you want to continue to work on lcdproc. However you will need to ask @willfe about this, I don't have access level to add new people, I think.

> I also took out the stuff about usbfs, as this is obviously old info (do you want me to do the same for the picolcd drivers docs)?

Of course, whenever you see an area of improvement creating a PR or opening an issue is welcome. Especially also for the documentation. It is so much more difficult writing documentation: Code you can always judge on whether it works.

An other area, where your expertise would help a lot, is issue #13 as I don't have any experience with libusb myself. (I have no idea if this is an easy or a difficult job. I will tag a release soon, to have ready before the upcoming debian freeze.)

> Sorry for leaving in one of commented out lines of code, at work we’re not allowed to delete any lines, and I tend to get into the habit of it.

I think this is what we have git for. Though indeed this line would not end up in git, if I squash commits.

> Finally, I’ve updated the Changelog.md and CREDITS.md files, apologies if this is not my place to do so, just thought it may be easier for you.

That's fine. However updating ChangeLog.md from PRs tends to cause merge conflicts when multiple PRs are under review at the same time.

> Oh, and could you squash my code into one commit, I ask as I’ve not actually done that before, and I have this horrible image of me accidently reverting it all by mistake, he he.

Sure, I will do this shortly. However a long as you keep a branch with your history of changes, such mishaps are easy to repair.

Thanks a lot for your work.

ajw107 commented:
Don’t worry about commit access, I’m happy doing it this way, if you are.  I’ll also have a look at the picolcd documentation and issue 13 tomorrow, if that is okay?  From a quick look at it, and just to clarify, you want me to look through the other drivers, and any that ONLY have support for LIBUSB0.1, to add support for LIBUSB1.0 (and LIBFTDI1 by the looks of it, no experience of that particular library, but I’ll give it a go)?
Alex

From: haraldg [mailto:notifications@github.com]
Sent: 27 December 2016 20:58
To: lcdproc/lcdproc <lcdproc@noreply.github.com>
Cc: Alexander Wood <alex@alex-wood.org.uk>; Author <author@noreply.github.com>
Subject: Re: [lcdproc/lcdproc] Add driver for the Futaba TOSD-5711BB VFDisplay on Elonex Artisan/Sca… (#33)


No need to be sorry or apologize for anything. This is good stuff and we all learn as we go and reviews almost always find ways to improve the code - which is why it is a good thing to have them. (Maybe I'm a bit biased from working on the linux kernel though.)

You seem to be reasonable and competent. We probably should give you commit access if you want to continue to work on lcdproc. However you will need to ask @willfe<https://github.com/willfe> about this, I don't have access level to add new people, I think.

I also took out the stuff about usbfs, as this is obviously old info (do you want me to do the same for the picolcd drivers docs)?

Of course, whenever you see an area of improvement creating a PR or opening an issue is welcome. Especially also for the documentation. It is so much more difficult writing documentation: Code you can always judge on whether it works.

An other area, where your expertise would help a lot, is issue #13<https://github.com/lcdproc/lcdproc/issues/13> as I don't have any experience with libusb myself. (I have no idea if this is an easy or a difficult job. I will tag a release soon, to have ready before the upcoming debian freeze.)

Sorry for leaving in one of commented out lines of code, at work we’re not allowed to delete any lines, and I tend to get into the habit of it.

I think this is what we have git for. Though indeed this line would not end up in git, if I squash commits.

Finally, I’ve updated the Changelog.md and CREDITS.md files, apologies if this is not my place to do so, just thought it may be easier for you.

That's fine. However updating ChangeLog.md from PRs tends to cause merge conflicts when multiple PRs are under review at the same time.

Oh, and could you squash my code into one commit, I ask as I’ve not actually done that before, and I have this horrible image of me accidently reverting it all by mistake, he he.

Sure, I will do this shortly. However a long as you keep a branch with your history of changes, such mishaps are easy to repair.

Thanks a lot for your work.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/lcdproc/lcdproc/pull/33#issuecomment-269381772>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AB-KPPsORT7WFeksCadLGkvf7vwTbg8yks5rMXvhgaJpZM4LQmqD>.

haraldg commented:
> I’ll also have a look at the picolcd documentation and issue 13 tomorrow, if that is okay?

Great.

> From a quick look at it, and just to clarify, you want me to look through the other drivers, and any that ONLY have support for LIBUSB0.1, to add support for LIBUSB1.0

Yes. If it is a lot of work or needs testing on real HW, then maybe just compile a list of drivers first and look on the Mailinglist for testers, before putting more effort into it. But lets disucss the details in the proper issue tracker.

> (and LIBFTDI1 by the looks of it, no experience of that particular library, but I’ll give it a go)?

I don't think there are many drivers using this and some patches are floating around, that did only the libftdi change without the libusb change. Turned out that this doesn't work, but the patches should be a good start.
